src/main/java/games/battlelore/BattleloreGameState.java

```diff
@@ -0,0 +1,284 @@
+package games.battlelore;
+
+import core.AbstractGameState;
+import core.AbstractParameters;
+import core.components.Component;
+import core.components.Deck;
+import core.components.GridBoard;
+import core.components.PartialObservableDeck;
+import core.interfaces.IGamePhase;
+import core.interfaces.IStateHeuristic;
+import games.GameType;
+import games.battlelore.cards.CommandCard;
+import games.battlelore.components.MapTile;
+import games.battlelore.components.Unit;
+
+import java.util.*;
+
+public class BattleloreGameState extends AbstractGameState {
+
+    public enum BattleloreGamePhase implements IGamePhase {
+        CommandAndOrderStep, //Player Plays One Command Card
+        MoveStep,
+        AttackStep,
+        VictoryPointStep,
+        DrawStep,
+        LoreStep
+        //Last three steps can be implemented later
+    }
+
+    public enum UnitType {
+        Decoy, BloodHarvester, ViperLegion, CitadelGuard, YeomanArcher;
+    }
+
+    int playerCount;
+    Random random;
+    BattleloreGameParameters parameters;
+
+    int numberOfRounds;
```

Comment:
Rounds and turns should be handled in a TurnOrder object, not in the game state.

Reply:
To clarify this as well: the TurnOrder object already keeps track of turns and rounds, so this is duplicating functionality that could be accessed from the TurnOrder (and/or adjusted as needed in the custom TurnOrder object, which at the moment adds nothing to the base class)